### ISSUE TITLE
Getting InaccessibleObjectException while using AnnotateNullable

### ISSUE THREAD

**Issue by Vyom-Yadav** [codespecs/daikon#381]
Hi, Team daikon,

We wish to integerate the Checker Framework with the Checkstyle project. After setting up Checker Framework, it recommended-

_Inference of [@Nullable](https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/Nullable.html): If your code uses the standard CLIMB-to-top default (Section [31.5.3](https://checkerframework.org/manual/#climb-to-top)) or the NonNull default, then use the [AnnotateNullable](http://plse.cs.washington.edu/daikon/download/doc/daikon.html#AnnotateNullable) tool of the Daikon invariant detector._

We are facing some problems while using the AnnotateNullable tool, we ran the following commands-
```bash
openjdk version "11.0.14" 2022-01-18
OpenJDK Runtime Environment (build 11.0.14+9-post-Debian-1)
OpenJDK 64-Bit Server VM (build 11.0.14+9-post-Debian-1, mixed mode, sharing)
```
```bash
Distributor ID:	Kali
Description:	Kali GNU/Linux Rolling
Release:	2022.1
Codename:	kali-rolling
```
```bash
$ mkdir daikonparent
$ cd daikonparent
$ wget http://plse.cs.washington.edu/daikon/download/daikon-5.8.10.tar.gz
$ tar zxf daikon-5.8.10.tar.gz
```
**Set the environment variables**
```
$ make -C $DAIKONDIR rebuild-everything
```
Output-
[log.txt](https://github.com/codespecs/daikon/files/8493698/log2.txt) (Downloadable format)
[gist](https://gist.github.com/Vyom-Yadav/36526cce3e992f3604fe6e87d7082768)

We wish to use the Daikon invariant detector only for the AnnotateNullable tool.

After that trying to follow the [commands mentioned under the annotate nullable](https://plse.cs.washington.edu/daikon/download/doc/daikon.html#AnnotateNullable) tool on a test file
```bash
$ java -cp $DAIKONDIR/daikon.jar daikon.Chicory \
--dtrace-file=an.dtrace.gz /home/vyom/IdeaProjects/ConfigCheckstyle/src/TestP/Test.java

Exception in thread "main" java.lang.reflect.InaccessibleObjectException: Unable to make field private static final java.lang.String com.sun.tools.javac.launCchhiecr.orMya iwnarn.binugn: dnlo ereNcaomrdes  wearcec perisnstiedbl
e: module jdk.compiler does not "opens com.sun.tools.javac.launcher" to unnamed module @701fc37a
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:340)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:280)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:176)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:170)
	at daikon.chicory.DaikonVariableInfo.addDeclVar(DaikonVariableInfo.java:771)
	at daikon.chicory.DaikonVariableInfo.addClassVars(DaikonVariableInfo.java:556)
	at daikon.chicory.RootInfo.enter_process(RootInfo.java:43)
	at daikon.chicory.Runtime.process_new_classes(Runtime.java:449)
	at daikon.chicory.Runtime.enter(Runtime.java:248)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.main(Main.java:119)
Warning: Target exited with 1 status
```
As a workaround for this we tried adding flag-
```
--add-opens jdk.compiler/com.sun.tools.javac.launcher=ALL-UNNAMED
```
```bash
$ java --add-opens jdk.compiler/com.sun.tools.javac.launcher=ALL-UNNAMED -cp $DAIKONDIR/daikon.jar daikon.Chicory \
--dtrace-file=an.dtrace.gz /home/vyom/IdeaProjects/ConfigCheckstyle/src/TestP/Test.java

Exception in thread "main" java.lang.reflect.InaccessibleObjectException: Unable to make field private static final java.lang.String com.sun.tools.javac.launcher.Main.bundlCeNhamice oarcy cewassrniibnleg::  mnood urlee cjodkr.dcso mwpeilreer  pdoreis nntoetd
 "opens com.sun.tools.javac.launcher" to unnamed module @701fc37a
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:340)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:280)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:176)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:170)
	at daikon.chicory.DaikonVariableInfo.addDeclVar(DaikonVariableInfo.java:771)
	at daikon.chicory.DaikonVariableInfo.addClassVars(DaikonVariableInfo.java:556)
	at daikon.chicory.RootInfo.enter_process(RootInfo.java:43)
	at daikon.chicory.Runtime.process_new_classes(Runtime.java:449)
	at daikon.chicory.Runtime.enter(Runtime.java:248)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.main(Main.java:119)
Warning: Target exited with 1 status
```
Still got the same error, also the error message was distorted-
`Exception in thread "main" java.lang.reflect.InaccessibleObjectException: Unable to make field private static final java.lang.String com.sun.tools.javac.launcher.Main.bundlCeNhamice oarcy cewassrniibnleg::  mnood urlee cjodkr.dcso mwpeilreer  pdoreis nntoetd`

Any help in getting Daikon invariant detector set up and successfully using Annotate Nullable tool will be appreciated.

Thanking You,
Vyom Yadav

**Comment by markro49** [comment]
Thank you for the detailed problem report.  However, I don't see source for /home/vyom/IdeaProjects/ConfigCheckstyle/src/TestP/Test.java so I can try to reproduce your results.

**Comment by Vyom-Yadav** [comment]
Hi @markro49 
Thanks for the response. I was getting the exception even when it was an empty class, or a simple POJO or anything, it was source independent. But still, I will share the what `Test.java` looks like-
```java
package TestP;

public class Test {
    private static class foo {
        private int i;

        private int getI() {
            return i;
        }

        private void setI(int i) {
            this.i = i;
        }
    }

    void foo() {
        foo obj = new foo();
        System.out.println(obj.i);
    }
}
```
Let me know if anything else is required :)

**Comment by markro49** [comment]
I'm sorry, I didn't read your command line carefully enough.  You don't run any of the Daikon tools on '.java' files; you run them on the compiled '.class' files.

I ran the test in my $DAIKONDIR. First, I edited Test.java to include a main method. Then I ran the follow commands:
 javac -g Test.java
 java -cp .:$DAIKONDIR/daikon.jar daikon.Chicory --dtrace-file=an.dtrace.gz Test
 java -cp $DAIKONDIR/daikon.jar daikon.Daikon an.dtrace.gz --no_text_output  --config java/daikon/annotate_nullable.config
 java -cp $DAIKONDIR/daikon.jar daikon.AnnotateNullable an.inv.gz > nullable-annotations.jaif

Here is my modified version of Test.java:
//package TestP;

public class Test {
    public static void main(String[] args) {
        foo();
    }

    private static class foo {
        private int i;

        private int getI() {
            return i;
        }

        private void setI(int i) {
            this.i = i;
        }
    }

    static void foo() {
        foo obj = new foo();
        System.out.println(obj.i);
    }
}
